controlplane/api/v1beta2/ck8scontrolplane_types.go

```diff
@@ -109,6 +116,26 @@ type CK8sControlPlaneMachineTemplate struct {
 	NodeDeletionTimeout *metav1.Duration `json:"nodeDeletionTimeout,omitempty"`
 }
 
+// RolloutStrategy describes how to replace existing machines
+// with new ones.
+type RolloutStrategy struct {
+	// rollingUpdate is the rolling update config params.
+	// +optional
+	RollingUpdate *RollingUpdate `json:"rollingUpdate,omitempty"`
+}
+
+// RollingUpdate is used to control the desired behavior of rolling update.
+type RollingUpdate struct {
+	// maxSurge is the maximum number of control planes that can be scheduled above or under the
+	// desired number of control planes.
+	// Value can be an absolute number 1 or 0.
+	// Defaults to 1.
+	// Example: when this is set to 1, the control plane can be scaled
+	// up immediately when the rolling update starts.
+	// +optional
+	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
```

Comment:
I honestly do not like that upstream also uses `intstr.IntOrString`, an `int` would enable us to have proper CRD validation. Do we know if we can use `kubebuilder:validation:XValidation` in anyway with `IntOrString`?

Reply:
Hello, I do not know as I am not so familiar with those annotations.